zebra/debug_nl.c

```diff
@@ -0,0 +1,1291 @@
+/*
+ * Copyright (c) 2018 Rafael Zalamena
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include <zebra.h>
+
+#ifdef NETLINK_DEBUG
+
+#include <sys/socket.h>
+
+#include <asm/types.h>
+#include <linux/if_arp.h>
+#include <linux/netlink.h>
+#include <linux/nexthop.h>
+#include <linux/rtnetlink.h>
+
+#include <stdio.h>
+#include <stdint.h>
+
+#include "zebra/rt_netlink.h"
+
+const char *nlmsg_type2str(uint16_t type)
```

Comment:
Can this and all other functions be moved into someplace in the library code? I can see them being generally useful.

Also, is it worth converting the big switch case into a number -> name lookup table?

Reply:
> Can this and all other functions be moved into someplace in the library code? I can see them being generally useful.

Sure thing, but I'd suggest people to move as they need otherwise we'd enlarge the binaries for no reason.

> Also, is it worth converting the big switch case into a number -> name lookup table?

I don't know... is it? I never measured the difference in code/binary.